scipy/optimize/_minimize_constrained.py

```diff
@@ -0,0 +1,485 @@
+        Slack variables at the solution. ``n_ineq`` and ``n_eq`` are the
+        total number of equality and inequality constraints.
+    niter : int
+        Total number of iterations.
+    nfev : int
+        Total number of objective function evaluations.
+    ngev : int
+        Total number of objective function gradient evaluations.
+    nhev : int
+        Total number of lagragian Hessian evaluations.
+    ncev : int
+        Total number of constraint evaluations.
+    njev : int
+        Total number of constraint Jacobian matrix evaluations.
+    niter : int
+        Total number of CG iterations.
+    cg_info : Dict
+        Dictionary containing information about the latest CG iteration:
+
+            - niter : Number of iterations.
+            - stop_cond : Reason for CG subproblem termination:
+
+                1. Iteration limit was reached;
+                2. Reached the trust-region boundary;
+                3. Negative curvature detected;
+                4. Tolerance was satisfied.
+
+            - hits_boundary : True if the proposed step is on the boundary
+              of the trust region.
+
+    execution_time : float
+        Total execution time.
+    trust_radius : float
+        Trust radius at the last iteration.
+    penalty : float
+        Penalty function at last iteration.
+    tolerance : float
+        Tolerance for barrier subproblem at the last iteration.
+        Exclusive for ``tr_interior_point``.
+    barrier_parameter : float
+        Barrier parameter at the last iteration. Exclusive for
+        ``tr_interior_point``.
+    status : {0, 1, 2, 3}
+        Termination status:
+
+            * 0 : The maximum number of function evaluations is exceeded.
+            * 1 : `gtol` termination condition is satisfied.
+            * 2 : `xtol` termination condition is satisfied.
+            * 3 : `callback` function requested termination.
+
+    message : str
+        Termination message.
+    method {"eq_constrained_sqp", "tr_interior_point"}
+        Optimization method used.
+    constr_violation : float
+        Constraint violation at last iteration.
+    optimality : float
+        Norm of the lagrangian gradient at last iteration.
+    fun : float
+        For the `eq_constraineq_sqp` method this is the objective function
+        evaluated at the solution and for the `tr_interior_point` method
+        this is the barrier  function evaluated at the solution.
+    grad : ndarray, (n,)
+        For the `eq_constraineq_sqp` method this is the gradient of the
+        objective function evaluated at the solution and for the
+        `tr_interior_point` method  this is the gradient of the barrier
+        function evaluated at the solution.
+    constr : ndarray, shape (n_ineq + n_eq,)
+        For the `eq_constraineq_sqp` method this is the equality constraint
+        evaluated at the solution and for the `tr_interior_point` method
+        this is he barrier function constraint.
```

Comment:
OK. And `constr_eq` and `constr_ineq` may have been modified somewhat from what the user might expect from the original problem formulation? For instance, some of the `constr_ineq` may have opposite sign?
As with the objective function, this is a situation where the user can get the residuals/slacks of the original constraints if he/she wants. Since the residuals/slacks of the original problem are not computed by the algorithm, I can see why it might not be worthwhile to do extra computation at every iteration.
But I can also see potential for conclusion. I'm not going to suggest a solution; just pointing this out.